resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtils.scala

```diff
@@ -451,15 +465,20 @@ trait MesosSchedulerUtils extends Logging {
   }
 
   /** Creates a mesos resource for a specific port number. */
-  private def createResourcesFromPorts(portsAndRoles: List[(Long, String)]) : List[Resource] = {
-    portsAndRoles.flatMap{ case (port, role) =>
-      createMesosPortResource(List((port, port)), Some(role))}
+  private def createResourcesFromPorts(
+       portsAndResourcesInfo: List[(Long, (String, AllocationInfo, Option[ReservationInfo]))])
+    : List[Resource] = {
+    portsAndResourcesInfo.flatMap{ case (port, rInfo) =>
+      createMesosPortResource(List((port, port)), Option(rInfo._1),
```

Comment:
Maybe a comment here on what exactly is going on?

Reply:
Sure.

Reply:
fixed